src/time_series_metadata_store.jl

```diff
@@ -0,0 +1,1004 @@
+    len = length(table)
+    if len == 0
+        return Dates.DateTime(Dates.Minute(0)), 0
+    elseif len > 1
+        throw(
+            InvalidValue(
+                "There are more than one sets of SingleTimeSeries initial times and lengths: $table",
+            ),
+        )
+    end
+
+    row = table[1]
+    return Dates.DateTime(row.initial_timestamp), row.length
+end
+
+# check_consistency is not implemented on StaticTimeSeries because new types may have
+# different requirments than SingleTimeSeries. Let future developers make that decision.
+
+function get_forecast_initial_times(store::TimeSeriesMetadataStore)
+    params = get_forecast_parameters(store)
+    isnothing(params) && return []
+    return get_initial_times(params.initial_timestamp, params.count, params.interval)
+end
+
+function get_forecast_parameters(store::TimeSeriesMetadataStore)
+    query = """
+        SELECT
+            horizon
+            ,initial_timestamp
+            ,interval_ms
+            ,resolution_ms
+            ,window_count
+        FROM $METADATA_TABLE_NAME
+        WHERE horizon IS NOT NULL
+        LIMIT 1
+        """
+    table = Tables.rowtable(_execute(store, query))
+    isempty(table) && return nothing
+    row = table[1]
+    return ForecastParameters(;
+        horizon = row.horizon,
+        initial_timestamp = Dates.DateTime(row.initial_timestamp),
+        interval = Dates.Millisecond(row.interval_ms),
+        count = row.window_count,
+        resolution = Dates.Millisecond(row.resolution_ms),
+    )
+end
+
+function get_forecast_window_count(store::TimeSeriesMetadataStore)
+    query = """
+        SELECT
+            window_count
+        FROM $METADATA_TABLE_NAME
+        WHERE window_count IS NOT NULL
+        LIMIT 1
+        """
+    table = Tables.rowtable(_execute(store, query))
+    return isempty(table) ? 0 : table[1].window_count
+end
+
+function get_forecast_horizon(store::TimeSeriesMetadataStore)
+    query = """
+        SELECT
+            horizon
+        FROM $METADATA_TABLE_NAME
+        WHERE horizon IS NOT NULL
+        LIMIT 1
+        """
+    table = Tables.rowtable(_execute(store, query))
+    return isempty(table) ? 0 : table[1].horizon
+end
+
+function get_forecast_initial_timestamp(store::TimeSeriesMetadataStore)
+    query = """
+        SELECT
+            initial_timestamp
+        FROM $METADATA_TABLE_NAME
+        WHERE horizon IS NOT NULL
+        LIMIT 1
+        """
+    table = Tables.rowtable(_execute(store, query))
+    return if isempty(table)
+        nothing
+    else
+        Dates.DateTime(table[1].initial_timestamp)
+    end
+end
+
+function get_forecast_interval(store::TimeSeriesMetadataStore)
+    query = """
+        SELECT
+            interval_ms
+        FROM $METADATA_TABLE_NAME
+        WHERE interval_ms IS NOT NULL
+        LIMIT 1
+        """
+    table = Tables.rowtable(_execute(store, query))
+    return if isempty(table)
+        Dates.Millisecond(0)
```

Comment:
This should be `nothing` by analogy with `get_forecast_initial_timestamp`, right?

Reply:
Sure. Changing all of these will causes some breakages in PSI, but we can deal with it.